### PR TITLE
UMD

### PR DESCRIPTION
**Pull request by dannytranlx**
Hi Marc,

I want to use your sets of behaviors on a RequireJS project. I added some wrappers. I did test on a CommonJS project and it looks like it is working also for a RequireJS project.

Is is still WIP because lines 6 to 8 (on lib/index.js) doesn't make sense at all. I am wondering if you could give me a hand to fix this to avoid put hard-coded paths.

Let me know what do you think!

Thanks!

### CODE REVIEW

**Comment by dannytranlx** [comment]
You can use ?w=1 to check the diff without whitespace :)

https://github.com/lemieux/marionette.behaviors/pull/4/files?w=1

File: lib/index.js [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
 
-module.exports = {
-    jQueryBehavior: require('./behaviors/jquery-behavior.js'),
-    StickitBehavior: require('./behaviors/stickit-behavior.js'),
-    bootstrapValidatorBehavior: require('./behaviors/bootstrapvalidator-behavior.js')
-};
+(function(root, factory) {
+    if (typeof define === 'function' && define.amd) {
+        define([
+            '/bower_components/marionette.behaviors/lib/behaviors/jquery-behavior.js',
```
Comment (lemieux) [inline lib/index.js]: these should be relatives path

Comment (lemieux) [inline lib/index.js]: I believe you can use the same paths as in the CommonJS requires, no?

Comment (dannytranlx) [inline lib/index.js]: Something must be wrong with our configs then. I will try on a blank RequireJS project. The relative path append to the current url and goes like `http://localhost/apps/hashhash/behaviors/jquery-behavior.js` :confused:

File: lib/behaviors/bootstrapvalidator-behavior.js [hunk 2]
```diff
@@ -1,30 +1,43 @@
...
+            require('backbone.marionette'),
+            require('underscore'),
+            require('bootstrapValidator')
+        );
+    }
+})(this, function(Marionette, _, BootstrapValidator) {
 
-require('bootstrapValidator');
+    return Marionette.Behavior.extend({
+        onRender: function() {
```
Comment (dannytranlx) [inline lib/behaviors/bootstrapvalidator-behavior.js]: Please note that I switched this `onShow` to `onRender` here. I don't remember why I used `onShow`, but it doesn't seem to break anything.

Comment (lemieux) [inline lib/behaviors/bootstrapvalidator-behavior.js]: onShow should be used when you need to compute stuff when it's in the dom (height, width of elements) which stickit doesn't need. Nice catch1

**Comment by dannytranlx** [comment]
I was able to use this lib as AMD package with this config:

```
packages: [
        {
            name: 'marionette-behaviors',
            location: '../bower_components/marionette.behaviors/lib',
            main: 'index'
        }
    ],
```

Not sure why it doesn't work with the regular path object:
```
paths: {
        // Bower components
       [...]
        'marionette-behaviors': '../bower_components/marionette.behaviors/lib/index'
}
```

Oh well, it should be ready to merge now. Please take note of the switch of `onShow` to `onRender` for the `bootstrapValidator-behavior`.

File: lib/behaviors/bootstrapvalidator-behavior.js (hunk 2 above)
Comment (dannytranlx) [inline lib/behaviors/bootstrapvalidator-behavior.js]: It's bootstrap validator!

Comment (lemieux) [inline lib/behaviors/bootstrapvalidator-behavior.js]: oh right... talking about bootstrap validator, it's no longer in npm!

Comment (dannytranlx) [inline lib/behaviors/bootstrapvalidator-behavior.js]: does it still work on your end? should we update to [OpenValidator](https://github.com/OpenValidator/openvalidator)?

Comment (lemieux) [inline lib/behaviors/bootstrapvalidator-behavior.js]: yup, what I should validate though is if FormValidator (new name of BootstrapValidator) changed the function to formValidator...

**Comment by lemieux** [comment]
all good?

**Comment by dannytranlx** [comment]
Yeah, ready to merge if it looks good to you!

**Comment by lemieux** [comment]
version bumped, npm and bower released!

**Pull request merged**